spark/src/main/java/org/datacleaner/spark/SparkJobLauncher.java

```diff
@@ -0,0 +1,119 @@
+ * Copyright (C) 2014 Neopost - Customer Information Management
+ *
+ * This copyrighted material is made available to anyone wishing to use, modify,
+ * copy, or redistribute it subject to the terms and conditions of the GNU
+ * Lesser General Public License, as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public License
+ * for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this distribution; if not, write to:
+ * Free Software Foundation, Inc.
+ * 51 Franklin Street, Fifth Floor
+ * Boston, MA  02110-1301  USA
+ */
+package org.datacleaner.spark;
+
+import java.io.InputStream;
+
+import org.apache.metamodel.DataContext;
+import org.apache.metamodel.csv.CsvDataContext;
+import org.apache.metamodel.util.HdfsResource;
+import org.apache.spark.SparkConf;
+import org.apache.spark.api.java.JavaRDD;
+import org.apache.spark.api.java.JavaSparkContext;
+import org.datacleaner.configuration.DataCleanerConfiguration;
+import org.datacleaner.configuration.JaxbConfigurationReader;
+import org.datacleaner.connection.Datastore;
+import org.datacleaner.job.AnalysisJob;
+import org.datacleaner.job.JaxbJobReader;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class SparkJobLauncher {
+    
+    private static final Logger logger = LoggerFactory.getLogger(SparkJobLauncher.class);
+
+    private final DataCleanerConfiguration _dataCleanerConfiguration;
+
+    public SparkJobLauncher(DataCleanerConfiguration dataCleanerConfiguration) {
+        _dataCleanerConfiguration = dataCleanerConfiguration;
+    }
+
+    public SparkJobLauncher(String confXmlPath) {
+        _dataCleanerConfiguration = readDataCleanerConfiguration(new HdfsResource(confXmlPath));
+    }
+
+    public void launchJob(AnalysisJob analysisJob) {
+        final String datastoreName = analysisJob.getDatastore().getName();
+
+        final Datastore datastore = _dataCleanerConfiguration.getDatastoreCatalog().getDatastore(datastoreName);
+        if (datastore == null) {
+            throw new IllegalStateException("Datastore referred by the job (" + datastoreName
+                    + ") has not been found in the specified DataCleanerConfiguration");
+        } else {
+            JavaSparkContext sc = null;
+            try {
+                DataContext dataContext = datastore.openConnection().getDataContext(); 
+                if (dataContext instanceof CsvDataContext) {
+                    CsvDataContext csvDataContext = (CsvDataContext) dataContext;
+                    String datastoreFilePath = csvDataContext.getResource().getQualifiedPath();
+            
+                    SparkConf conf = new SparkConf().setAppName("DataCleaner-spark").setMaster("yarn-client");
```

Comment:
I think it's recommended not to set master like this. Isn't it usually given externally from the launcher?